util/types/json/functions.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/juju/errors"
+)
+
+// Type returns type of JSON as string.
+func (j JSON) Type() string {
+	switch j.typeCode {
+	case typeCodeObject:
+		return "OBJECT"
+	case typeCodeArray:
+		return "ARRAY"
+	case typeCodeLiteral:
+		switch byte(j.i64) {
+		case jsonLiteralNil:
+			return "NULL"
+		default:
+			return "BOOLEAN"
+		}
+	case typeCodeInt64:
+		return "INTEGER"
+	case typeCodeFloat64:
+		return "DOUBLE"
+	case typeCodeString:
+		return "STRING"
+	default:
+		msg := fmt.Sprintf(unknownTypeCodeErrorMsg, j.typeCode)
+		panic(msg)
+	}
+}
+
+// Extract receives several path expressions as arguments, matches them in j, and returns:
+//  ret: target JSON matched any path expressions. maybe autowrapped as an array.
+//  found: true if any path expressions matched.
+func (j JSON) Extract(pathExprList ...string) (ret JSON, found bool, err error) {
```

Comment:
We do not need to process the pathExprList string list  for every json object.
For example `select json_extract(json_col, '$.a', '$.c') from test_json`. We should preprocess the `'$.a', '$.c'` to some structure and apply it to each row from test_json table.
So I think we should define a structure to represent the processed pathExpr list and pass the preprocessed pathExpr list to Json.Extrac() function.